js/everything.js

```diff
@@ -364,56 +371,71 @@ function toggleElement(event, toggleID, buttonText) {
     }
 }
 
+/**
+ * wrapParseYouTubeVideoID the first v query value.
+ * Will return null if there's no v query param
+ * @return {string|null}
+ */
 function getCurrentVideoID() {
-    var v = getParameterByName(window.location.search, "v");
-    // If the URL had 2 v parameters, try parsing the second (usually when ?v=someurl&v=xyz)
-    var vParams = window.location.search.match(/v=\w+/g);
-    if (vParams && vParams.length > 1) {
-        v = vParams[vParams.length - 1].replace("v=", "");
+    var v = URI(window.location).search(true).v;
+    if (!v) {
+        return null;
     }
-    else if (v.length > 1) {
-        return wrapParseYouTubeVideoID(v);
+
+    if (Array.isArray(v)) {
+        v = v[0];
     }
-    return v;
+
+    return wrapParseYouTubeVideoID(v);
 }
 
+/**
+ * Return the current times position, provided by the t query param, parsed to seconds.
+ * @returns {*}
+ */
 function getCurrentTimePosition() {
-    var t = parseInt(getParameterByName(window.location.search, "t"), 10);
+    var t = parseInt(URI(window.location).search(true).t, 10);
     if (t > 0 && t < Number.MAX_VALUE) {
         return t;
     }
     return 0;
 }
-
+/**
+ * Return the q query param if one exists.
+ * @returns {boolean|string|*|Array}
+ */
 function getCurrentSearchQuery() {
-    var q = getParameterByName(window.location.search, "q");
-    return q;
-}
-
-function removeSearchQueryFromURL(url) {
-    if (window.location.search.length !== 0) {
-        url = window.location.href.replace(window.location.search, "");
-    }
-    return url;
+    return URI(window.location).search(true).q;
 }
 
+/**
+ * Return the current URL, appending v=videoID and t=videoPosition, if set.
+ * Remove hashes from the URL.
+ * @param {string} videoID
+ * @param {number} [videoPosition]
+ * @returns {string}
+ */
 function makeListenURL(videoID, videoPosition) {
-    var url = removeSearchQueryFromURL(window.location.href);
-    // Remove any #s which break functionality
-    url = url.replace("#", "");
-    url += "?v=" + videoID;
+    var url = URI(window.location);
+    url.setSearch("v", videoID);
+
     if (videoPosition) {
-        url += "&t=" + videoPosition;
+        url.addSearch("t", videoPosition);
```

Comment:
why not `setSearch("t", videoPosition)`? is there any different between those 2 functions?